e2e/cypress/integration/plugins/demo_plugin/webhook_spec.js

```diff
@@ -0,0 +1,232 @@
+        cy.postMessage('some message');
+
+        // * Verify message is posted in the demo channel
+        cy.visit(`/${testTeam.name}/channels/demo_plugin`);
+        cy.findAllByTestId('postView').contains(`MessageHasBeenPosted: @${testUser.username}, ~Town Square`);
+    });
+
+    it('MM-T2408/3 - User joined a channel Webhook event', () => {
+        // # Join channel
+        cy.visit(`/${testTeam.name}/channels/${testChannel}`);
+        cy.wait(TIMEOUTS.TWO_SEC);
+
+        // * Verify message is posted in the demo channel
+        cy.visit(`/${testTeam.name}/channels/demo_plugin`);
+        cy.findAllByTestId('postView').contains(`UserHasJoinedChannel: @${testUser.username}, ~my_test_channel`);
+    });
+
+    it('MM-T2408/4 - ​User left a channel Webhook event', () => {
+        // # Leave channel
+        cy.visit(`/${testTeam.name}/channels/${testChannel}`);
+        cy.wait(TIMEOUTS.TWO_SEC);
+        cy.uiLeaveChannel(false);
+
+        // * Verify message is posted in the demo plugin channel
+        cy.visit(`/${testTeam.name}/channels/demo_plugin`);
+        cy.findAllByTestId('postView').contains(`UserHasLeftChannel: @${testUser.username}, ~my_test_channel`);
+    });
+
+    it('MM-T2408/5/6 - User edited a message Webhook event', () => {
+        // # Post message
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+        cy.wait(TIMEOUTS.TWO_SEC);
+        cy.postMessage('some message');
+
+        // # Get last post ID
+        cy.getLastPostId().then((postID) => {
+            // # click  dot menu button
+            cy.clickPostDotMenu();
+
+            // # click edit post
+            cy.get(`#edit_post_${postID}`).click();
+
+            // # Edit message to 'some message add'
+            cy.get('#edit_textbox').
+                should('be.visible').
+                and('be.focused').
+                wait(TIMEOUTS.HALF_SEC).
+                type(' add');
+
+            // # Click button Edit
+            cy.get('#editButton').click();
+
+            // # Open demo plugin channel
+            cy.visit(`/${testTeam.name}/channels/demo_plugin`);
+            cy.wait(TIMEOUTS.TWO_SEC);
+
+            // * Verify event posted in the channel
+            cy.findAllByTestId('postView').contains(`MessageHasBeenUpdated: @${testUser.username}, ~Town Square`);
+        });
+    });
+
+    it('MM-T2408/7 - User adds a reaction to a message Webhook event', () => {
+        // # Post message
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+        cy.wait(TIMEOUTS.TWO_SEC);
+        cy.postMessage('some message');
+
+        cy.getLastPostId().then((postId) => {
+            // # Click the add reaction icon
+            cy.clickPostReactionIcon(postId);
+
+            // # Choose "slightly_frowning_face" emoji
+            // delaying 500ms in case of lag
+            // eslint-disable-next-line cypress/no-unnecessary-waiting
+            cy.get('.emoji-picker__items #emoji-1f641').wait(500).click();
```

Comment:
Should we `wait` on `cy.clickPostReactionIcon` instead?

Reply:
This seems to be the standard pattern of choosing the emojis